core/domain/classifier_services.py

```diff
@@ -100,8 +101,39 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    state = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+        'TextInput']['classifier_id']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
```

Comment:
No, but you have a registry from which you can get the classes. It currently returns instances of the class but I'm not sure why that can't be changed.

That said, I think one of the potential sources of confusion here is that your naming is not really distinguishing the object storing the classifier data ("Classifier" in this file) from the LDA classifier object itself (which I guess is "sc"). We need better names so that we can communicate without ambiguity. Perhaps the former should be called ClassifierData (not sure) and the latter is Classifier (and there are different types of Classifiers)? Then we can refer to a Classifier class and a Classifier instance without ambiguity about what these things mean.